lib/storage/src/content_manager/snapshots/recover.rs

```diff
@@ -229,11 +231,28 @@ async fn _do_recover_from_snapshot(
                     *shard_id,
                     cancel::CancellationToken::new(),
                 )
-                .await?;
+                .await;
 
-            if !recovered {
-                log::debug!("Shard {} is not in snapshot", shard_id);
-                continue;
+            match recovered {
+                Ok(recovered) => {
+                    if !recovered {
+                        log::debug!("Shard {} is not in snapshot", shard_id);
+                        continue;
+                    }
+                }
+                Err(err) => {
+                    log::error!("Failed to recover shard {}: {err}", shard_id);
+                    // delete whole collection dir if it was created to not leave a valid config behind
+                    if collection_did_not_exist {
+                        let collection_path = toc.get_collection_path(&collection_pass.to_string());
+                        log::debug!(
+                            "Deleting collection directory for failed snapshot recovery: {:?}",
+                            collection_path
+                        );
+                        tokio::fs::remove_dir_all(&collection_path).await?;
```

Comment:
I don't think we can do this in distributed environment. Consider cluster of 3 machines: one machine restores snapshot, 2 other create empty collection. If snapshot recovery fails, we only remove collection from one machine, 2 other machines will still think it exists.